cinema/views.py

```diff
@@ -24,21 +30,29 @@
 class GenreViewSet(viewsets.ModelViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
```

Comment:
instead of duplicating code, you can set DEFAULT_PERMISSION_CLASSES and DEFAULT_AUTHENTICATION_CLASSES in REST_FRAMEWORK constant in settings.py and override these attributes where needed